### PR TITLE
fix for that motioned in https://github.com/tesseract-ocr/tesseract/issues/3680

### PR DESCRIPTION
**Pull request by leleliu008**
#3680 

Reference:
https://github.com/Kitware/CMake/blob/master/Modules/Platform/Android-Determine.cmake
https://android.googlesource.com/platform/ndk/+/master/build/cmake/android.toolchain.cmake
https://developer.android.com/ndk/guides/cpu-features#the_google_cpu_features_library
https://github.com/google/cpu_features#android-ndks-drop-in-replacement

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -721,6 +721,22 @@ if (WIN32 AND CLANG AND OPENMP_BUILD)
...
 
+# fix for that motioned in https://github.com/tesseract-ocr/tesseract/issues/3680
+#
+# 2021-12-18 14:56:11 UTC  leleliu008@gmail.com
+#
+# Reference:
+# https://github.com/Kitware/CMake/blob/master/Modules/Platform/Android-Determine.cmake
+# https://android.googlesource.com/platform/ndk/+/master/build/cmake/android.toolchain.cmake
+# https://developer.android.com/ndk/guides/cpu-features#the_google_cpu_features_library
+# https://github.com/google/cpu_features#android-ndks-drop-in-replacement
```
Comment (stweil) [inline CMakeLists.txt:732]:
```suggestion
```

Comment (stweil) [reply CMakeLists.txt:732]: Usually we don't add such details to the code, only short comments where needed. All that information is also in the commit message, so it does not get lost.

**Comment by zdenop** [comment]
thanks

**Comment by amitdo** [comment]
I don't think we should add `-DANDROID`.

https://stackoverflow.com/questions/15328751/android-macro-suddenly-not-defined

https://github.com/neoml-lib/neoml/issues/30

**Pull request merged**